azure/services/tags/tags.go

```diff
@@ -57,31 +59,47 @@ func (s *Service) Reconcile(ctx context.Context) error {
 	defer done()
 
 	for _, tagsSpec := range s.Scope.TagsSpecs() {
+		existingTags, err := s.client.GetAtScope(ctx, tagsSpec.Scope)
+		if err != nil {
+			return errors.Wrap(err, "failed to get existing tags")
+		}
+		tags := make(map[string]*string)
+		if existingTags.Properties != nil && existingTags.Properties.Tags != nil {
+			tags = existingTags.Properties.Tags
+		}
+
+		if !s.isResourceManaged(tags) {
+			s.Scope.V(4).Info("Skipping tags reconcile for not managed resource")
+			continue
+		}
+
 		annotation, err := s.Scope.AnnotationJSON(tagsSpec.Annotation)
 		if err != nil {
 			return err
 		}
-		changed, created, deleted, newAnnotation := tagsChanged(annotation, tagsSpec.Tags)
+		changed, createdOrUpdated, deleted, newAnnotation := tagsChanged(annotation, tagsSpec.Tags)
 		if changed {
 			s.Scope.V(2).Info("Updating tags")
-			result, err := s.client.GetAtScope(ctx, tagsSpec.Scope)
-			if err != nil {
-				return errors.Wrap(err, "failed to get existing tags")
-			}
-			tags := make(map[string]*string)
-			if result.Properties != nil && result.Properties.Tags != nil {
-				tags = result.Properties.Tags
-			}
-			for k, v := range created {
-				tags[k] = to.StringPtr(v)
+			if len(createdOrUpdated) > 0 {
```

Comment:
nit: not going to cause any huge performance difference, but I feel (update + delete) may be costlier than createAndUpdate except maybe in cases when there are large number of user created tags, then we'll be sending all those values across the network.

Reply:
Yes, 2 calls vs 1 call was what I was meaning to say.